### PR TITLE
fix(npm): resolve node_modules dir relative to package.json instead of cwd

### PR DESCRIPTION
**Pull request by dsherret**

### CODE REVIEW

File: cli/args/config_file.rs [hunk 1]
```diff
@@ -524,6 +535,14 @@ impl ConfigFile {
     /// Filenames that Deno will recognize when discovering config.
     const CONFIG_FILE_NAMES: [&str; 2] = ["deno.json", "deno.jsonc"];
 
+    // todo(dsherret): in the future, we should enforce all callers
+    // to provide a resolved path
```
Comment (dsherret) [inline cli/args/config_file.rs:539]: What I've done in other projects is introduced a `ResolvedPath` struct, then force resolving that once and passing it around.

File: cli/tests/testdata/run/with_package_json/no_deno_json/sub_dir/main.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import * as path from "../../../../../../../test_util/std/path/mod.ts";
+
+const parentDir = path.dirname(path.dirname(path.fromFileUrl(import.meta.url)));
+
+console.log(parentDir);
+console.log(Deno.statSync(parentDir));
```
Comment (dsherret) [inline cli/tests/testdata/run/with_package_json/no_deno_json/sub_dir/main.js:6]: Oops...

File: cli/args/config_file.rs (hunk 1 above)
Comment (bartlomieju) [reply cli/args/config_file.rs:539]: That's a very good idea, we should do the same

File: test_util/src/lib.rs [hunk 3]
```diff
@@ -2328,6 +2341,37 @@ pub fn parse_max_mem(output: &str) -> Option<u64> {
...
+      })?;
+    } else if file_type.is_file() {
+      std::fs::copy(&new_from, &new_to).with_context(|| {
+        format!("Copying {} to {}", new_from.display(), new_to.display())
+      })?;
+    }
+  }
+
+  Ok(())
+}
```
Comment (dsherret) [inline test_util/src/lib.rs:2373]: This was taken from elsewhere in the CLI.

**Review by bartlomieju — APPROVED** [review]
LGTM, great improvement. Thanks for looking into this!

**Pull request merged**